### DISCUSSION TITLE
Seeking help with querying nested data

### DISCUSSION THREAD

**Discussion by ChrisB007** [prisma/prisma#10638] [Q&A] [answered]
H

})
```

**Comment by thrant** [comment, marked as answer]
See https://www.prisma.io/docs/concepts/components/prisma-client/select-fields#include-relations-and-select-relation-fields

You are missing the "include" parameter from the save, which defines if you will return just the Audience record or also some of it's related records.  The below will include the SingleAudience, and further include the Options and SingleCosts related to it. The same Include param exists in all queries (findUnique, findMany, create, update, ...)

```
const includes = {
    SingleAudience: {
         include {
            Options: true,
            SingleCost: true
        }
}
const audience = await prisma.audience.create({
      data: req.body,
      include: includes
    });
```

Note that the returned object will look like this, i.e. it does not return an array directly but an object with an array:

```
Audience: {
  id: '....',
  audiences: [  
  {
    id: 1,
    section: "Gender",
    options: [
      { gender: "Men", checked: false },
      { gender: "Women", checked: false },
      { gender: "Anyone", checked: true }
    ],
    costs: [{ cost: "0", costTier: "Standard", checked: true }],
    description:
      "This describes the single audience."
  },
  {
    id: 2,
    section: "Age",
    options: [
    ....
```

**Reply by ChrisB007** [reply]
Thank you so much. I appreciate your resonse

**Reply by ChrisB007** [reply]
@thrant Sorry to bother you again. I have a pressing question about my model. I am following the Prisma doc about creating a 1 - 1 relation, but when I tried doing the same - I didn't get any errors,  just that additional fields were created in my model that I don't understand. Do you know what it does?

```
// Prisma example - from their docs

One-to-one relations
One-to-one (1-1) relations refer to relations where at most one record can be connected on both sides of the relation. In the example below, there's one 1-1-relation between User and Profile:

Relational databases
MongoDB
model User {
  id      Int      @id @default(autoincrement())
  profile Profile?
}

model Profile {
  id     Int  @id @default(autoincrement())
  user   User @relation(fields: [userId], references: [id])
  userId Int // relation scalar field (used in the `@relation` attribute above)
}
The userId relation scalar is a direct representation of the foreign key in the underlying database.This 1-1-relation expresses the following:

"a user can have zero or one profiles" (because the profile field is optional on User)
"a profile must always be connected to one user"

```



```
// this got added to my model (one user has one home)
model User{
  id               Int      @id @default(autoincrement())
  isUser           @default("false")
  createdAt        DateTime @default(now())
  updatedAt        DateTime @updatedAt
  home             Home? @relation("userTouser", fields: [userId], references: [id])
  userIdId         Int?
  user             user[] @relation("userTouser")
  home             home[]
 }

model Home {
  id              Int      @id @default(autoincrement())
  type            String
  home_id         Int
  user            User   @relation(fields: [home_id], references: [id])
  createdAt       DateTime @default(now())
  updatedAt       DateTime @updatedAt
}

```

```
// what was created in my model that wasn't in Prisma examples

 home             Home? @relation("userTouser", fields: [userId], references: [id])
  userIdId         Int?
  user             user[] @relation("userTouser")
  home             home[]

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
